File: src/4-Problem09.pas
program Turunan;
uses crt;

{ Kamus }
var
  derajat: integer; { derajat polinom }
  polinom: array [1..100] of integer; { koefisien polinom }
  i: integer; { pengulangan }

{ Algoritma }
begin
  { Membaca masukan }
  write('Masukkan pangkat paling besar : '); readln(derajat);
  for i := derajat downto 0 do begin
    write('Masukkan koefisien x^2 : '); readln(polinom[i]);
  end;
  
  { Menghitung turunan }
  for i := 0 to derajat-1 do
    polinom[i] := (i+1) * polinom[i+1];

  polinom[derajat] := 0;
  derajat := derajat - 1;

  { Menuliskan polinom }
  write('Maka hasil turunannya adalah ');
  for i := derajat downto 0 do begin
    write(polinom[i], 'x^', i);
    if (i > 0) then
      write(' + ');
  end;
  writeln;

end.

File: 4-Problem09.py
def inputKoefisien(N):
	kata="Masukkan koefisien x^"+str(N)+" : "
	arr=input(kata)
	return (int(arr))



n=input("Masukkan pangkat paling besar : ")
N=int(n)
pangkatPertama=N
array=[]
while (pangkatPertama>=0):
	arr=inputKoefisien(pangkatPertama)
	array.append(int(arr))
	pangkatPertama=pangkatPertama-1

pangkat=N
hasil=''
while (pangkat>0):
	isi=array[0]*pangkat
	array.pop(0)
	pangkat=pangkat-1
	if (pangkat == N-1):
		hasil=hasil+"Maka hasilnya adalah "+str(isi)+"x^"+str(pangkat)+" "
	else:
		hasil=hasil+"+ "+str(isi)+"x^"+str(pangkat)+" "
print (hasil)
		
